Assets/BossRoom/Scripts/Client/UI/Lobby/JoinLobbyUI.cs

```diff
@@ -0,0 +1,172 @@
+using System.Collections.Generic;
+using BossRoom.Scripts.Shared.Infrastructure;
+using BossRoom.Scripts.Shared.Net.UnityServices.Lobbies;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BossRoom.Scripts.Client.UI
+{
+    /// <summary>
+    /// Handles the list of LobbyPanelUIs and ensures it stays synchronized with the lobby list from the service.
+    /// </summary>
+    public class JoinLobbyUI : ObserverBehaviour<LobbyServiceData>
+    {
+        [SerializeField] private LobbyPanelUI m_LobbyPanelPrototype;
+        [SerializeField] private InputField m_JoinCodeField;
+        [SerializeField] private CanvasGroup m_CanvasGroup;
+
+        private IInstanceResolver m_Container;
+
+        /// <summary>
+        /// Key: Lobby ID, Value Lobby UI
+        /// </summary>
+        private readonly Dictionary<string, LobbyPanelUI> m_LobbyButtons = new Dictionary<string, LobbyPanelUI>();
+
+        private LobbyServiceData m_LobbyServiceData;
+        private LobbyUIMediator m_LobbyUIMediator;
+
+        private readonly Dictionary<string, LocalLobby> m_LocalLobby = new Dictionary<string, LocalLobby>();
+
+        private UpdateRunner m_UpdateRunner;
+
+        private void Awake()
+        {
+            m_LobbyPanelPrototype.gameObject.SetActive(false);
+        }
+
+        private void OnDisable()
+        {
+            if (m_UpdateRunner != null)
+            {
+                m_UpdateRunner.Unsubscribe(PeriodicRefresh);
+            }
+        }
+
+        [Inject]
+        private void InjectDependencies(IInstanceResolver container, LobbyUIMediator lobbyUIMediator, LobbyServiceData lobbyServiceData, UpdateRunner updateRunner)
+        {
+            m_Container = container;
+            m_LobbyUIMediator = lobbyUIMediator;
+            m_LobbyServiceData = lobbyServiceData;
+            m_UpdateRunner = updateRunner;
+
+            m_UpdateRunner.Subscribe(PeriodicRefresh, 10f);
+            BeginObserving(m_LobbyServiceData);
+        }
+
+        public void OnJoinButtonPressed()
+        {
+            m_LobbyUIMediator.JoinLobbyWithCodeRequest(m_JoinCodeField.text.ToUpper());
+        }
+
+        private void PeriodicRefresh(float _)
+        {
+            m_LobbyUIMediator.QueryLobbiesRequest(false);
+        }
+
+        public void OnRefresh()
+        {
+            m_LobbyUIMediator.QueryLobbiesRequest(true);
+        }
+
+        protected override void UpdateObserver(LobbyServiceData observed)
```

Comment:
I couldn't find logic on pagination, only that we're limiting the number of lobbies received using a const. If I didn't miss it, what do you think of adding a next step (not for this iteration) and adding pagination handling? Is it something other samples are educating on? It's not a trivial thing, both client and server side, it might be worth it educating on this.

Reply:
It's a more advanced use. I didn't see an example of Pagination handling in the lobby sample.

Reply:
created https://jira.unity3d.com/browse/MTT-2626